ReactAndroid/src/main/java/com/facebook/react/modules/core/Timing.java

```diff
@@ -107,13 +120,84 @@ public void doFrame(long frameTimeNanos) {
     }
   }
 
+  private class IdleFrameCallback implements Choreographer.FrameCallback {
+
+    @Override
+    public void doFrame(long frameTimeNanos) {
+      if (isPaused.get()) {
+        return;
+      }
+
+      // If the JS thread is busy for multiple frames we cancel any other pending runnable.
+      if (mCurrentIdleCallbackRunnable != null) {
+        mCurrentIdleCallbackRunnable.cancel();
+      }
+
+      mCurrentIdleCallbackRunnable = new IdleCallbackRunnable(frameTimeNanos);
+      getReactApplicationContext().runOnJSQueueThread(mCurrentIdleCallbackRunnable);
```

Comment:
This is to avoid queueing multiple calls to JS if the JS thread is blocked for multiple frames. What I'm doing here is every frame I'm sending a new runnable and cancelling the previous one so only the latest will be sent to JS.